webrender/src/frame_builder.rs

```diff
@@ -843,12 +844,35 @@ impl FrameBuilder {
             line.color = shadow_color.premultiplied();
             let mut info = info.clone();
             info.rect = info.rect.translate(&shadow_offset);
+
+            // Since the shadow items share the same clip with original elements,
+            // an offset is needed for correct clipping; Therefore, instead of
+            // creating an additional clip node, we adjust the local clip rect
+            // in the LayerPrimitiveInfo to perform the right clipping.
+            let (_, shadow_clip) = self.shadow_prim_stack[idx].1[0];
+            let fast_clip_and_scroll;
+            if clip_and_scroll == shadow_clip {
+                fast_clip_and_scroll = clip_and_scroll;
+            } else {
+                let node = clip_scroll_tree.nodes.get(&clip_and_scroll.clip_node_id()).unwrap();
+                let mut clip_rect = node.local_viewport_rect.translate(&shadow_offset);
```

Comment:
Hrm. This feels a bit wrong to me. How do we know that the ClipScrollNode has the same coordinate system as the shadow? What about the other clips in the clip hierarchy? I'm not sure this is the correct approach here, but it could be that I simply don't understand the problem. I worry that this is merely hiding the overall issue.

Doesn't it make sense to render